src/app/projects/states/dashboard/directives/task-dashboard/task-dashboard.coffee

```diff
@@ -44,6 +44,20 @@ angular.module('doubtfire.projects.states.dashboard.directives.task-dashboard',
         taskFilesUrl: TaskFeedback.getTaskFilesUrl($scope.task)
       }
 
+      # add the target grade based on the index:
+      # 0: pass, 1: credit and so on
+      switch $scope.task.project().target_grade
+        when 0
+          $scope.task.project().target_grade_word = "Pass"
+        when 1
+          $scope.task.project().target_grade_word = "Credit"
+        when 2
+          $scope.task.project().target_grade_word = "Distinction"
+        when 3
+          $scope.task.project().target_grade_word = "High Distinction"
+        else
+          $scope.task.project().target_grade_word = "Unknown"
+      
```

Comment:
I think there's a reson the ` $scope.task.project().target_grade ` is index. so I created a new variable.

Reply:
Can you move this into the **project-service.coffee** in the `addProjectMethods` function? This is where we are currently updating the project objects to have the methods we need. 

There is also an array of grade names in the grades service that we can use.

Make this a function something like:

```coffee
project.targetGradeWord = () -> gradeService.grades[switch $scope.task.project().target_grade]
```

You could add some array protection to make sure the grade is valid.

Reply:
Hi @macite there you go [refactor: move the index to target grade function to project-service](https://github.com/doubtfire-lms/doubtfire-web/pull/505/commits/81fab51826e8fed66d6e89fb3b1d91bc52a70044)